### PR TITLE
feat: hook up color and theme variables

### PR DESCRIPTION
**Pull request by tinaszheng**
image

# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
The focus of this PR is to update UI styles, switch components, and color configurations in the UI demo app.

### Detailed summary
- Updated UI styles for various components like `PhotoUpload`, `TopNav`, `AuthCardWrapper`, `CodePreview`, `ThemeSwitch`, `Styling`
- Changed color configurations and added a brand color
- Removed `Switch` component in `TopNav`
- Updated `tailwind.config.ts` with a new brand color
- Modified CSS classes for `Switch` components in different files
- Adjusted primary and secondary color variables in `globals.css`
- Added logic to set primary color in `Providers` based on config
- Updated class names and styles in various components to match the new UI styles

> The following files were skipped due to too many changes: `examples/ui-demo/src/components/configuration/Styling.tsx`, `examples/ui-demo/src/components/icons/illustration-style.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: examples/ui-demo/src/app/page.tsx [hunk 1]
```diff
@@ -27,7 +27,7 @@ export default function Home() {
 
   return (
     <main
-      className={`flex bg-gray-50 flex-col h-screen ${publicSans.className}`}
+      className={`flex bg-gray-50 dark:bg-gray-800 flex-col h-screen ${publicSans.className}`}
```
Comment (tinaszheng) [inline examples/ui-demo/src/app/page.tsx:30]: very sad attempt at initial dark mode for the app - waiting on figma from julia

File: examples/ui-demo/src/components/configuration/Styling.tsx [hunk 2]
```diff
@@ -112,7 +113,7 @@ export function CornerRadiusOptions() {
         <button
           className={`${
             option.className
-          } py-2 flex-1 basis-0 bg-[#EFF4F9] text-fg-accent-brand hover:opacity-80 ${
+          } py-2 flex-1 basis-0 bg-[#EFF4F9] text-[#363FF9] hover:opacity-80 ${
```
Comment (tinaszheng) [inline examples/ui-demo/src/components/configuration/Styling.tsx:116]: we should probably copy `fg-accent-brand` into another variable separate from the css variable that could be overridden in the demo app

File: examples/ui-demo/tailwind.config.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { withAccountKitUi } from "@account-kit/react/tailwind";
 import type { Config } from "tailwindcss";
 
 const config = {
-  darkMode: ["class"],
+  darkMode: "selector",
```
Comment (moldy530) [inline examples/ui-demo/tailwind.config.ts:5]: oh nice! does this ultimately allow you to do dark mode on sub trees or does this just enable using `dark` as a class?

**Review by moldy530 — APPROVED** [review]

File: examples/ui-demo/tailwind.config.ts (hunk 3 above)
Comment (tinaszheng) [reply examples/ui-demo/tailwind.config.ts:5]: no, this just replaces the "class" version in latest tailwind:
https://tailwindcss.com/docs/dark-mode#toggling-dark-mode-manually

**Pull request merged**